internal/controlplane/handlers_repositories.go

```diff
@@ -343,17 +345,30 @@ func (s *Server) ListRemoteRepositoriesFromProvider(
 		}
 		p, err := providers.GetProviderBuilder(ctx, provider, s.store, s.cryptoEngine, &s.cfg.Provider, pbOpts...)
 		if err != nil {
-			return nil, status.Errorf(codes.Internal, "cannot get provider builder: %v", err)
+			zerolog.Ctx(ctx).Error().Err(err).Msg("cannot get provider builder")
+			erroringProviders = append(erroringProviders, provider.Name)
+
+			// skip over this provider
+			continue
 		}
 
 		results, err := s.listRemoteRepositoriesForProvider(ctx, provider.Name, p, projectID)
 		if err != nil {
-			return nil, err
+			zerolog.Ctx(ctx).Error().Err(err).Msg("cannot list repositories for provider")
+			erroringProviders = append(erroringProviders, provider.Name)
+
+			// skip over this provider
+			continue
 		}
 
 		out.Results = append(out.Results, results...)
 	}
 
+	// If all providers failed, return an error
+	if len(erroringProviders) > 0 && len(out.Results) == 0 {
+		return nil, util.UserVisibleError(codes.Internal, "cannot list repositories for providers: %v", erroringProviders)
```

Comment:
The logic LGTM, the main caveat I see is that folks will now miss some error messages that were intended to be guidance. e.g. `util.UserVisibleError(codes.PermissionDenied, "cannot get credential for provider: did you run `minder provider enroll`?")` is now lost.

Reply:
That's true, I can add the "did you run `minder provider enroll`?" back. With the new provider implementation there shouldn't be a scenario where people have a provider without a credential, so it would only happen when a user registered in the past and didn't enroll a provider, and then they return later. I figured that wasn't common enough to keep the message.

Reply:
Let's keep this for now and fix later. I was thinking already of other providers